x-pack/filebeat/input/httpjson/pagination.go

```diff
@@ -103,6 +103,7 @@ func (p *pagination) newPageIterator(stdCtx context.Context, trCtx *transformCon
 		trCtx:      trCtx,
 		resp:       resp,
 		isFirst:    true,
+		n:          -1, // the first page needs to be 0 for every interval
```

Comment:
If `pageIterator.getPage` is rearranged slightly, this comment and assignment becomes unnecessary
```
func (iter *pageIterator) getPage() (*response, error) {
	bodyBytes, err := ioutil.ReadAll(iter.resp.Body)
	if err != nil {
		return nil, err
	}
	iter.resp.Body.Close()

	var r response
	r.header = iter.resp.Header
	r.url = *iter.resp.Request.URL
	r.page = iter.n
	iter.n++

	if len(bodyBytes) > 0 {
		if iter.pagination.decoder != nil {
			err = iter.pagination.decoder(bodyBytes, &r)
		} else {
			err = decode(iter.resp.Header.Get("Content-Type"), bodyBytes, &r)
		}
		if err != nil {
			return nil, err
		}
	}

	return &r, nil
}
```

Reply:
Good catch, will do that but will keep the comment in the function just in case someone moves a couple of lines in the future.